pep.rst

```diff
@@ -0,0 +1,627 @@
+
+    # define a DOM type...
+
+    def html(*args: str | Thunk) -> DOM:
+        for arg in args:
+            match arg:
+                case str():
+                    # parse arg in the context of an HTML template that
+                    # is being built
+                case getvalue, _, _, _:
+                    # interpolate `getvalue()` in this HTML template
+
+As a named tuple, thunks can be matched using a case statement either as a tuple
+(as above), or with respect to the class ``Thunk`` and any desired attributes to
+be structurally unpacked.
+
+.. note::
+
+    For more complete details on how to implement this fully using
+    ``html.parser`` in the `stdlib
+    https://docs.python.org/3/library/html.parser.html`_, see the companion
+    tutorial PEP.
+
+With an implemented ``html`` tag, it is then possible to write code like the
+following:
+
+.. code-block:: python
+
+    from typing import Any
+    from my_htmllib import html, DOM
+
+    def title(report: str, props: dict[str, Any], styling: dict[str, Any]) -> DOM:
+        return html'<div {props} style={styling}>Report: {report}</div>'
+
+    report = 'Profit & Loss'  # arbitrary, so need to interpolate properly
+    styling = {'color': 'blue'}  # for a CSS style tag, special interpolation required
+    props = {FIXME, but include a boolean element}
+    dom = title(report, props, styling)
+
+It is also possible to recursively compose tag strings:
+
+    FIXME todolist
+
+Tag names like ``html`` bind to a callable, or **tag function**. Such tag
+functions can use logic specific to the DSL it is working with, such as HTML and
+any interpolation rules. In addition, because tag strings generalize f-strings,
+it is possible to use Python in the construction in the template, given that any
+interpolated expressions are Python expressions.
+
+Specification
+=============
+
+In the rest of this specification, ``mytag`` will be used for an arbitrary tag.
+
+Grammar
+-------
+
+The tag name can be any **undotted** name that isn't an existing valid string or
+bytes prefix, as seen in the `lexical analysis specification
+https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals`_:
+
+.. code-block:: text
+
+    stringprefix: "r" | "u" | "R" | "U" | "f" | "F"
+                : | "fr" | "Fr" | "fR" | "FR" | "rf" | "rF" | "Rf" | "RF"
+
+    bytesprefix: "b" | "B" | "br" | "Br" | "bR" | "BR" | "rb" | "rB" | "Rb" | "RB"
+
+As with other string literals, no whitespace can be between the tag and the
+quote mark.
+
+.. note::
+
+    The restriction to use undotted names can be relaxed to dotted names in the
+    future, if there is a compelling usage.
+
+No string concatenation
+-----------------------
+
+Tag string concatenation isn't supported, which is `unlike other string literals
+https://docs.python.org/3/reference/lexical_analysis.html#string-literal-concatenation`_.
+
+.. note::
+
+    Tthe expectation is that triple quoting is sufficient. If string
```

Comment:
s/Tthe/The/

Reply:
Thanks! I added cspell/[Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) to my workflow.